rust/nix-store/src/store.rs

```diff
@@ -41,7 +41,10 @@ pub struct Store {
     context: Context,
 }
 impl Store {
-    pub fn open(url: &str) -> Result<Self> {
+    pub fn open<'a, 'b>(
+        url: &str,
+        params: impl IntoIterator<Item = (&'a str, &'b str)>,
```

Comment:
this was `params: &HashMap<&str, &str>` before, which was the correct type, i think. but i guess, in this case the zero cost abstraction outweighs correctness. a user with a  `&[(&str, &str)]` or even a `BTreeMap` does not need to pay the cost of converting to a `HashMap`.